packages/common/src/hooks/usePremiumContentAccess.ts

```diff
@@ -0,0 +1,60 @@
+import { PremiumConditions, Track } from 'models'
+import { useMemo } from 'react'
+
+import { useSelector } from 'react-redux'
+import { cacheUsersSelectors } from 'store/cache'
+import { premiumContentSelectors } from 'store/premium-content'
+import { CommonState } from 'store/reducers'
+import { Nullable } from 'utils'
+
+const { getUsers } = cacheUsersSelectors
+const { getPremiumTrackSignatureMap } = premiumContentSelectors
+
+export const usePremiumContentAccess = (track: Nullable<Partial<Track>>) => {
+  if (!track) {
+    return { isUserAccessTBD: false, doesUserHaveAccess: true }
+  }
+
+  const premiumTrackSignatureMap = useSelector(getPremiumTrackSignatureMap)
+
+  const { isUserAccessTBD, doesUserHaveAccess } = useMemo(() => {
+    const trackId = track.track_id
+    const isPremium = track.is_premium
+    const hasPremiumContentSignature =
+      !!track.premium_content_signature ||
+      !!(trackId && premiumTrackSignatureMap[trackId])
+    const isCollectibleGated = !!track.premium_conditions?.nft_collection
+    const isSignatureToBeFetched =
+      isCollectibleGated &&
+      !!trackId &&
+      premiumTrackSignatureMap[trackId] === undefined
+
+    return {
+      isUserAccessTBD: !hasPremiumContentSignature && isSignatureToBeFetched,
```

Comment:
are there any ways we could model this state more accurately? i.e. are there impossible combinations of `isUserAccessTBD` and `doesUserHaveAccess` (tbd: true & haveAccess: true?). could we model this with a string union something like `accessState: unknown | accessible | denied`?